### ISSUE TITLE
Allow boot from volume

### ISSUE THREAD

**Issue by jprovaznik** [redhat-openstack/openshift-on-openstack#304]
For bigger deployments it would be nice to have an option to use "boot from volume" nova option when creating instances instead of passing an image.

**Comment by ioggstream** [comment]
I got the "boot from volume" templates. Any idea about using `conditionals` for unifying the templates? @jprovaznik

**Comment by ioggstream** [comment]
@jprovaznik this feature could create issues when replacing nodes, and should be addressed in some way.

## heat can't manage boot-volumes when replacing host

- create an host with a boot volume using the following template https://pastebin.com/quB7Ky2j
- mark the host unhealthy
- update the stack

results in 
```
UPDATE_FAILED  Resource CREATE failed: 
BadRequest: resources.host: Invalid volume: volume 'xxx' status must be 'available'. 
Currently in 'in-use' (HTTP 400) (Request-ID: req-yyy)
```

**Comment by jprovaznik** [comment]
@ioggstream I wonder whether it's a feature or a race issue on heat side - I would expect that replacing node means deleting the node first (which should make volume available again) and then creating a new node.

**Comment by jprovaznik** [comment]
@ioggstream I don't think that there is much active development on this project so I wouldn't expect this (or other features) to be done though.

**Comment by ioggstream** [comment]
@jprovaznik I wrote here because I didn't find a relevant issue on trello. It's fine to continue the discussion there and forward port the stuff on the new openshift-ansible (I already made 2 PR).


 afaik heat workflow is the following:

1- create new server and ensure it's online
2- remove the old one
I'm on irc  #heat looking for insights now.

**Comment by jprovaznik** [comment]
@ioggstream I'm not on this project for a while, but with ceph-ansible I suppose you can define customized playbook for replacing a node (which does exactly what you want - so first deletes the node, then creates a new one) if it turns out that heat creates a new one before deleting the old one.

**Comment by ioggstream** [comment]
@jprovaznik can you point me to somebody interested to that work? I'd really like that any effort ccould be useful to a future solution in openshift-ansible. Thx++

**Comment by jprovaznik** [comment]
@ioggstream this is a really old RFE which came from testing scalability of these templates. IIRC the problem was that by default nova instance boots from image, which means that this image is copied from glance to a nova compute node where the instance is being started - if you boot hundreds of nodes at the same time, it will be bottleneck. You can ping @jeremyeder from the scalability team or openshift-ansible active developers, I think @tomassedovic works on openstack integration so he might navigate you to the right people.
